### PR TITLE
zxing-cpp: add version 1.3.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zxing-cpp/***

- support gcc >= 11 (adding additional patches)
- another test_package.cpp for zing-cpp/1.3.0
- add find_package(stb) in test_package

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
```
The system cannot execute the specified program.
ERROR: zxing-cpp/1.3.0 (test package): Error in test() method, line 24
	self.run(bin_path, run_environment=True)
	ConanException: Error 1 while executing bin\test_package
```

Oh.I can't find the way to solve this.
It will take some time to solve it as I currently do not have an environment to use Visual Studio C++.

**Comment by uilianries** [comment]
@toge Maybe @madebr and @SpaceIm know about, they contributed to this recipe before.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/zxing-cpp/all/conanfile.py [hunk 1]
```diff
@@ -66,13 +65,11 @@ def validate(self):
...
-            )
-        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) >= "11":
-            raise ConanInvalidConfiguration(
-                "zxing-cpp doesn't support gcc >= 11. Contributions are "
-                "welcome if you want to fix the build."
+                "This compiler is too old. {} needs a compiler with c++14 support".format(self.name)
             )
+        # FIXME: This is a workaround for "The system cannot execute the specified program."
+        if tools.Version(self.version) == "1.3.0" and is_msvc_static_runtime(self) and self.settings.build_type == "Debug":
+            raise ConanInvalidConfiguration("{}/{} doesn't support MT + Debug.".format(self.name, self.version))
```
Comment (madebr) [inline recipes/zxing-cpp/all/conanfile.py:72]: It is this combination wou want to exclude from the build matrix?
Can you reproduce the problem locally? Is it VS2017-only, or does it happen for all VC's?

The error you're trying to work around ("The system cannot execute the specified program.`) looks weird.
This log? https://github.com/conan-io/conan-center-index/pull/11210#issuecomment-1157258162

**Comment by prince-chrismc** [comment]
Seems like an older issue https://github.com/conan-io/conan-center-index/pull/8373#issuecomment-1068082478

Lots of projects struggle with MT since so few use it. I think this is a great improvement!

**Comment by madebr** [comment]
So this actually means we're either using zxing-cpp/stb_image wrong in the test_package,
or there is a bug in one of these libraries?

**Comment by axxel** [comment]
I'm the maintainer of the zxing-cpp library and just noticed this packaging effort. I'd like to point out a few things I noticed:
 * to build the lib we require c++17 since v1.2.0 (minimum VS 2019 16.8 / gcc 7 / clang 5)
 * I just released v1.4.0 a couple of days ago
 * up until 1.4 a c++11 compiler was sufficient to use the library (on the client side)

Question: when would you consider removing support to package 1.0.8 (which is totally outdated by now)?

**Comment by toge** [comment]
@axxel 
Thank you for your valuable comments.
I will reflect your comments in a separate PR with the addition of version 1.4.0. (https://github.com/conan-io/conan-center-index/pull/11614)

As for 1.0.8, in accordance with conan's version policyI intended to remove it after some time.
If you have any problems with it, please let me know why and I will remove it with a comment.

**Comment by axxel** [comment]
Very nice. I did not see the pending PR already adding 1.4. Looks good. Minor issue: the description seems to still claim it to be a "c++14" port.

**Pull request merged**